assets/scripts/sentiment/SentimentSurveyContainer.jsx

```diff
@@ -36,7 +36,9 @@ function SentimentSurveyContainer (props) {
         setVisible(true)
       }, SURVEY_DELAY_BEFORE_APPEAR)
     }
-  })
+    // only re-run the effect if the value of isDismissed has changed
```

Comment:
This took me a minute to get to, more info here:
https://reactjs.org/docs/hooks-effect.html#tip-optimizing-performance-by-skipping-effects

When I ran the code using debugger, it hit this hook several times. "By default, it runs both after the first render and after every update"

In this case, I belive we update the DOM several times. At least two I can think of: 

1)inital load with spinny load thing
2)Showing all the street stuff

There is probably a third one in there two given how many times I kept seeing the dialog box. 

Using it this way, we kind of 'short-circuit' the check. This would be the flow: 
* has isDismissed changed? If so: 
    * but isDismissed is now false, so do not set another timeout

This feels kinda hacky, but it does mean only a single timeout should get set. My biggest complaint with this solution is that it's not super obvious that it's there to prevent a bug.